op-e2e/interop/system2.go

```diff
@@ -0,0 +1,408 @@
+package interop
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	bss "github.com/ethereum-optimism/optimism/op-batcher/batcher"
+	batcherFlags "github.com/ethereum-optimism/optimism/op-batcher/flags"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/interopgen"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/fakebeacon"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/geth"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/opnode"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/setuputils"
+	"github.com/ethereum-optimism/optimism/op-node/node"
+	"github.com/ethereum-optimism/optimism/op-node/p2p"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/sync"
+	l2os "github.com/ethereum-optimism/optimism/op-proposer/proposer"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	"github.com/ethereum-optimism/optimism/op-service/endpoint"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+// rather than pass around a string for rpc, pass around an rpc interface
+// this interface should be able to have a simple version
+// and a copmlex version
+// a utility in the middle can decide which to use based on an interface check,
+// plus preferences if they can be applied
+type rpcClient interface {
+}
+
+// ignore unused variable warning
+var _ rpcClient = &rpc.Client{}
+
+// example: ask for a chainID plus the actor,
+// get the actor back
+// system2 should provide a complex rpcClient
+// or could return the simple rpcClient url
+
+// TestInterop stands up a basic L1
+// and multiple L2 states
+
+// system2 is a struct for creating resources
+// rather than make a big-bang of resources like system 1,
+// system 2 will create resources in a more modular way
+// and allows calls
+type system2 struct {
+	recipe          *interopgen.InteropDevRecipe
+	logger          log.Logger
+	hdWallet        *devkeys.MnemonicDevKeys
+	worldDeployment *interopgen.WorldDeployment
+	worldOutput     *interopgen.WorldOutput
+	beacon          *fakebeacon.FakeBeacon
+	l1              *geth.GethInstance
+	l2s             map[string]l2Set
+	clients         map[string]*ethclient.Client
+}
```

Comment:
The members of this struct are just what I found was needed for the creation steps while decomposing the function. They might not all be needed.